Content.Shared/Buckle/SharedBuckleSystem.Buckle.cs

```diff
@@ -360,6 +361,9 @@ private void Buckle(Entity<BuckleComponent> buckle, Entity<StrapComponent> strap
         else if (user != null)
             _adminLogger.Add(LogType.Action, LogImpact.Low, $"{ToPrettyString(user):player} buckled {ToPrettyString(buckle)} to {ToPrettyString(strap)}");
 
+        if (TryComp<PullableComponent>(strap.Owner, out var pullable) && pullable.Puller == buckle.Owner)
+            _pullingSystem.TryStopPull(strap.Owner, pullable);
```

Comment:
I think you'll need to change `TryStopPull` to have some sort of "forced" behavior where it doesn't allow it to fail. Presently, the current code doesn't handle a situation where the `TryStopPull` call fails to actually stop the pull (since it's not guaranteed) causing the assert to still trigger.